### ISSUE TITLE
Is it possible to resume training from a saved checkpoint?

### ISSUE THREAD

**Issue by issam9** [UKPLab/sentence-transformers#1605]
Hi, i'm trying to resume training from checkpoints that i save while finetuning a sentence transformer. I tried the following but it starts training from scratch.

```
model = SentenceTransformer('checkpoint_path')
model.fit([(loader, loss)])
```

**Comment by nreimers** [comment]
This is sadly currently not supported

**Comment by steve-marmalade** [comment]
Came here looking for this, so I figured I'd leave a +1. Something equivalent to the `resume_from_checkpoint` flag in the huggingface [transformers.Trainer.train](https://huggingface.co/docs/transformers/main_classes/trainer#transformers.Trainer.train) would be awesome.

Is this a high level of effort to implement?

**Comment by jaideep11061982** [comment]
@nreimers  at least if we do load from a saved best ckps , then it should have updated weights to start from epoch 0 ?

**Comment by david-waterworth** [comment]
@nreimers any plans on implementing this? It's critical when training on sagemaker using spot instances - the training script must be able to be interruptible as sagemaker will stop/start it at will based on availability (and using spot instances is much cheaper).

**Comment by tomaarsen** [comment]
Yes @david-waterworth, the intention is to include this in the upcoming Trainer refactor in #2449.

- Tom Aarsen

**Comment by NRamirez01** [comment]
@tomaarsen My open PR adds resume_to_checkpoint to fit() in fitmixin.py which is what ollama uses at least for SentenceTransformersFinetuneEngine
